azurelinuxagent/ga/exthandlers.py

```diff
@@ -1068,55 +1070,79 @@ def is_responsive(heartbeat_file):
     def launch_command(self, cmd, timeout=300, extension_error_code=1000, env=None):
         begin_utc = datetime.datetime.utcnow()
         self.logger.verbose("Launch command: [{0}]", cmd)
+
         base_dir = self.get_base_dir()
 
-        if env is None:
-            env = {}
-        env.update(os.environ)
+        with tempfile.TemporaryFile(dir=base_dir, mode="w+b") as stdout:
```

Comment:
Something to discuss - I would recommend redirecting the stdout/stderr in the `/var/log/azure/(EXTN)/` directory - If in case the file becomes too large, the user should be able to delete the file forcefully, and need not be root.

Reply:
Discussed offline - the log directory is also restricted to root (for writes), and also deleting the file has no effect since the extension still holds a handle to it. When the extension completes, the file will be deleted by the OS.